src/openforms/authentication/signals.py

```diff
@@ -59,19 +63,43 @@ def set_auth_attribute_on_session(
         )
         raise PermissionDenied(_("Demo plugins require an active admin session."))
 
+    # TODO check this log is accurate
     logger.debug(
         "Persisting form auth to submission %s. Plugin: '%s' (setting attribute '%s')",
         instance.uuid,
         plugin,
         attribute,
     )
 
-    store_auth_details(instance, form_auth)
-
-
-@receiver(submission_complete, dispatch_uid="auth.clean_submission_form_auth")
-def clean_submission_form_auth(sender, request: Request, **kwargs):
-    if FORM_AUTH_SESSION_KEY not in request.session:
-        return
-
-    del request.session[FORM_AUTH_SESSION_KEY]
+    if registrator_subject:
+        # we got registrator_subject data so form_auth is an employee
+        if registrator_subject.get("skipped_subject_info"):
+            # If continued without extra details, store the employee details in AuthInfo
+            store_auth_details(instance, form_auth)
+        else:
+            # If KVK/BSN is given, store those details in the AuthInfo model and the employee details in the Registrator model.
+            auth_save = {
+                "value": registrator_subject["value"],
+                "attribute": registrator_subject["attribute"],
+                # TODO we don't have a plugin to define here? things break if this doesn't exist (like the logout view)
+                "plugin": registrator_subject.get("plugin", "registrator"),
+            }
+            registrator_save = {
+                "value": form_auth["value"],
+                "attribute": form_auth["attribute"],
+                "plugin": form_auth["plugin"],
+            }
+            # TODO the "machtigingen" key dispeared
```

Comment:
can you elaborate?

Reply:
If there is a `machtigingen` key/value in the `form_auth` it won't get copied here. I don't think it is relevant here though.

Reply:
yeah you're right - maybe detect if there's a non-empty value and emit a `logger.warning(...)` in that case that the data was discarded. It shouldn't happen, but if it _does_ happen, we at least make the problem visible.